packages/block-library/src/legacy-widget/edit/form.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { store as noticesStore } from '@wordpress/notices';
+import { __ } from '@wordpress/i18n';
+import {
+	useEffect,
+	useRef,
+	useState,
+	useCallback,
+	forwardRef,
+	RawHTML,
+} from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { Button } from '@wordpress/components';
+
+export default function Form( { id, idBase, instance, setInstance } ) {
+	const ref = useRef();
+
+	const { html, setFormData } = useForm( {
+		id,
+		idBase,
+		instance,
+		setInstance,
+	} );
+
+	const onSubmit = useCallback(
+		( event ) => {
+			event.preventDefault();
+			if ( id ) {
+				setFormData( serializeForm( ref.current ) );
+			}
+		},
+		[ id ]
+	);
+
+	const onChange = useCallback(
+		debounce( () => {
+			if ( idBase ) {
+				setFormData( serializeForm( ref.current ) );
+			}
+		}, 300 ),
+		[ idBase ]
+	);
+
+	return (
+		<div className="widget open">
+			<div className="widget-inside">
+				<ObservableForm
+					ref={ ref }
+					className="form"
+					method="post"
+					onSubmit={ onSubmit }
+					onChange={ onChange }
+				>
+					<RawHTML className="widget-content">{ html }</RawHTML>
+					{ id && (
+						<Button type="submit" isPrimary>
+							{ __( 'Save' ) }
+						</Button>
+					) }
+				</ObservableForm>
+			</div>
+		</div>
+	);
+}
+
+function useForm( { id, idBase, instance, setInstance } ) {
+	const isStillMounted = useRef();
```

Comment:
Nit: Default it to `false`.
```suggestion
	const isStillMounted = useRef( false );
```